### PR TITLE
refactor(components): [affix] use JSX in Unit test

### PR DESCRIPTION
**Pull request by buqiyuan**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/affix/__tests__/affix.spec.tsx [hunk 1]
```diff
@@ -1,17 +1,14 @@
-import { nextTick } from 'vue'
+import { nextTick, type VNode } from 'vue'
```
Comment (sxzz) [inline packages/components/affix/__tests__/affix.spec.tsx:1]: Please import type individually:

```ts
import type { VNode } from 'vue'
```

put this line after line#4

File: packages/components/affix/__tests__/affix.spec.tsx [hunk 2]
```diff
@@ -1,17 +1,14 @@
...
 import { mount } from '@vue/test-utils'
 import { defineGetter, makeScroll } from '@element-plus/test-utils'
 import Affix from '../src/affix.vue'
 
 let clientHeightRestore: () => void
 
-const _mount = (template: string) =>
+const _mount = (template: VNode) =>
```
Comment (sxzz) [inline packages/components/affix/__tests__/affix.spec.tsx:8]:
```suggestion
const _mount = (render: () => VNode) =>
```

File: packages/components/affix/__tests__/affix.spec.tsx [hunk 3]
```diff
@@ -1,17 +1,14 @@
...
   mount(
     {
-      components: {
-        'el-affix': Affix,
-      },
-      template,
+      setup: () => () => template,
```
Comment (sxzz) [inline packages/components/affix/__tests__/affix.spec.tsx:11]:
```suggestion
      setup: () => render,
```

File: packages/components/affix/src/affix.vue [hunk 4]
```diff
@@ -126,4 +124,8 @@ onMounted(() => {
 useEventListener(scrollContainer, 'scroll', onScroll)
 useResizeObserver(root, () => update())
 useResizeObserver(target, () => update())
+
+defineExpose({
+  update,
```
Comment (sxzz) [inline packages/components/affix/src/affix.vue:129]: add
```ts
/**
  * @description xxx
*/
```

File: packages/components/affix/__tests__/affix.spec.tsx [hunk 5]
```diff
@@ -2,16 +2,14 @@ import { nextTick } from 'vue'
 import { mount } from '@vue/test-utils'
 import { defineGetter, makeScroll } from '@element-plus/test-utils'
 import Affix from '../src/affix.vue'
+import type { VNode } from 'vue'
 
 let clientHeightRestore: () => void
 
-const _mount = (template: string) =>
+const _mount = (render: () => VNode) =>
```
Comment (JeremyWuuuuu) [inline packages/components/affix/__tests__/affix.spec.tsx:9]: We should name this `renderer`

File: packages/components/affix/src/affix.vue [hunk 6]
```diff
@@ -98,15 +96,15 @@ const onScroll = () => {
...
-    fixed: state.fixed,
+    scrollTop: scrollTop.value,
+    fixed: fixed.value,
   })
 }
 
 watch(
-  () => state.fixed,
+  () => fixed.value,
   () => {
```
Comment (JeremyWuuuuu) [inline packages/components/affix/src/affix.vue:106]:
```ts
watch(
  () => fixed.value,
  (fixed) => {
    emit('change', fixed)
  }
)
```

File: packages/components/affix/__tests__/affix.spec.tsx (hunk 5 above)
Comment (JeremyWuuuuu) [reply packages/components/affix/__tests__/affix.spec.tsx:9]: I didn't see any special case in your code, can we simply this to:

```ts
const _mount = (template: VNode) => {
  return mount(template, options)
}
```

Comment (buqiyuan) [reply packages/components/affix/__tests__/affix.spec.tsx:9]: This is a question about code style, we should be consistent, what do you think？@sxzz

Comment (JeremyWuuuuu) [reply packages/components/affix/__tests__/affix.spec.tsx:9]: Different scenario we should write different code, in both cases I am fine with them, but you should avoid writing unnecessary code though, in this case the `setup` is completely redundant to me.

Comment (buqiyuan) [reply packages/components/affix/__tests__/affix.spec.tsx]: Yeah, you are right!

File: packages/components/affix/src/affix.vue (hunk 6 above)
Comment (buqiyuan) [reply packages/components/affix/src/affix.vue]: done

File: packages/components/affix/__tests__/affix.spec.tsx [hunk 7]
```diff
@@ -2,19 +2,13 @@ import { nextTick } from 'vue'
...
-    {
-      components: {
-        'el-affix': Affix,
-      },
-      template,
-    },
-    { attachTo: document.body }
-  )
+const _mount = (template: VNode) => {
+  return mount(() => template, { attachTo: document.body })
```
Comment (JeremyWuuuuu) [inline packages/components/affix/__tests__/affix.spec.tsx:10]: Can we remove this `() => template` and just use `mount(template)` here? Didn't see any difference here.

Comment (JeremyWuuuuu) [reply packages/components/affix/__tests__/affix.spec.tsx:10]: I got no comments expect this one for now. The rest of this change seems legit to me.

Comment (buqiyuan) [reply packages/components/affix/__tests__/affix.spec.tsx:10]: I want too, but `mount ` first argument must be `() => VNode`

image

File: packages/components/affix/__tests__/affix.spec.tsx [hunk 8]
```diff
@@ -2,19 +2,13 @@ import { nextTick } from 'vue'
...
-  mount(
-    {
-      components: {
-        'el-affix': Affix,
-      },
-      template,
-    },
-    { attachTo: document.body }
-  )
+const _mount = (template: VNode) => {
```
Comment (sxzz) [inline packages/components/affix/__tests__/affix.spec.tsx:9]:
```suggestion
const _mount = (render: () => VNode) => {
```

File: packages/components/affix/src/affix.vue [hunk 9]
```diff
@@ -98,15 +96,15 @@ const onScroll = () => {
...
+    scrollTop: scrollTop.value,
+    fixed: fixed.value,
   })
 }
 
 watch(
-  () => state.fixed,
-  () => {
-    emit('change', state.fixed)
+  () => fixed.value,
```
Comment (sxzz) [inline packages/components/affix/src/affix.vue:105]:
```suggestion
  fixed,
```

**Review by sxzz — APPROVED** [review]

**Pull request merged**